airgun/entities/lifecycleenvironment.py

```diff
@@ -34,14 +34,28 @@ def create(self, values, prior_entity_name=None,):
         else:
             self.create_environment_path(values)
 
-    def read(self):
+    def read(self, entity_name):
+        """Read specific lifecycle environment details from its Edit page"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        return view.read()
+
+    def read_all(self):
+        """Read all available lifecycle environments details from generic
+        lifecycle environments page
+        """
         view = self.navigate_to(self, 'All')
         return view.read()
 
     def update(self, values, entity_name='Library'):
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
         return view.fill(values)
 
+    def search_package(self, entity_name, package_name, cv_name=None):
+        """Search for specific package inside lifecycle environment"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.packages.search(package_name, cv=cv_name)
+        return view.packages.table.read()
+
 
```

Comment:
add `search_puppet_module`? or we do not need it?

Reply:
we don't need `search_*` methods to fetch associated packages/puppet_modules/etc, `read()` returns them. All the remaining t2/t3 tests are fetching what's associated, there's no need in search_* methods.
`search_package` is rather exclusion as there is customer's defect which affected searching of packages so separate tests which target package searching were added.